### PR TITLE
[APP-2816] Appc badge

### PR DESCRIPTION
**Pull request by Culurio**
**What does this PR do?**

This PR makes the APPC banner inside the App View clickable and redirects users to the bonus screen

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] See by commit

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [APP-2816](https://aptoide.atlassian.net/browse/APP-2816)

**What are the relevant tickets?**

  Tickets related to this pull-request: [APP-2816](https://aptoide.atlassian.net/browse/APP-2816)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new keys, etc.) 

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[APP-2816]: https://aptoide.atlassian.net/browse/APP-2816?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[APP-2816]: https://aptoide.atlassian.net/browse/APP-2816?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app-games/src/main/java/com/aptoide/android/aptoidegames/search/presentation/SearchView.kt
Comment (GuiFPires) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/search/presentation/SearchView.kt:1]: Missing the navigate here on LargeAppItem

**Review by GuiFPires — CHANGES_REQUESTED** [review]
There are two other use cases that might be worth taking a look (better to check with João/Ana as well)

- If we are on the bonus view already, clicking on the badge infinitely will lead to a loop
- A similar case happens with the apkfy bottom sheet, as the sheet isn't dismissed (I don't know if it should be or not, as when we click the app icon it is) when we click the badge, we can navigate infinite times to the bonus page

File: app-games/src/main/java/com/aptoide/android/aptoidegames/categories/presentation/CategoryDetailView.kt
Comment (GuiFPires) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/categories/presentation/CategoryDetailView.kt:1]: Also missing navigate on LargeAppItem here

**Review by GuiFPires — CHANGES_REQUESTED** [review]

**Review by GuiFPires — APPROVED** [review]

File: app-games/src/main/java/com/aptoide/android/aptoidegames/appview/AppViewScreen.kt [hunk 1]
```diff
@@ -292,10 +294,17 @@ fun AppViewContent(
       )
     }
     if (app.isAppCoins) {
+      val bonusTag = "bonus-banner"
+      val bonusTitle = "Bonus"
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/appview/AppViewScreen.kt:298]: Hardcoded string. Probably should be either translated recourse or fetched from network

File: app-games/src/main/java/com/aptoide/android/aptoidegames/appview/AppViewScreen.kt [hunk 2]
```diff
@@ -292,10 +294,17 @@ fun AppViewContent(
       )
     }
     if (app.isAppCoins) {
+      val bonusTag = "bonus-banner"
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/appview/AppViewScreen.kt:297]: I see these string literals repeated 7 times and all are connected with `getBonusIconLeft`.
Let's extract those as a reusable component to encapsulate these constants and not repeat them.

**Review by G0retZ — CHANGES_REQUESTED** [review]

**Review by GuiFPires — CHANGES_REQUESTED** [review]
If we open an appview via deeplink and the app is closed, the app crashes due to the lateinit vars of bundleTitle and bundleTag not being initialized for the BonusData

File: app-games/src/main/java/com/aptoide/android/aptoidegames/appview/AppViewScreen.kt [hunk 3]
```diff
@@ -293,10 +296,17 @@ fun AppViewContent(
       )
     }
     if (app.isAppCoins) {
+      val bonusViewModel = BonusViewModel()
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/appview/AppViewScreen.kt:299]: You should not instantiate view models on recompositons.

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/BonusSectionView.kt [hunk 4]
```diff
@@ -47,6 +47,8 @@ fun BonusSectionView(
   bundle: Bundle,
   navigate: (String) -> Unit,
 ) {
+  val bonusViewModel = BonusViewModel()
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/BonusSectionView.kt:50]: You should not instantiate view models on recompositons.

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/BonusViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    title: String,
+    tag: String
+  ) {
+    viewModelScope.launch {
+      BonusData.setBonusData(title, tag)
+    }
+  }
+
+  @Composable
+  fun rememberBonusData(): Pair<String, String> {
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/BonusViewModel.kt:21]: This is a wrong way to inject and use view model.
Refer the other view models and remember functions syntax and usage.

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/BonusViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import com.aptoide.android.aptoidegames.feature_apps.data.BonusData
+import kotlinx.coroutines.launch
+
+class BonusViewModel : ViewModel() {
+
+  fun addBundleData(
+    title: String,
+    tag: String
+  ) {
+    viewModelScope.launch {
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/BonusViewModel.kt:15]: Why do you need a view model scope here?

**Review by G0retZ — CHANGES_REQUESTED** [review]

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/BonusSectionView.kt [hunk 7]
```diff
@@ -47,6 +47,8 @@ fun BonusSectionView(
...
+  bonusViewModel.addBundleData(bundle.title, bundle.tag)
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/BonusSectionView.kt:51]: As well as you should not call view model functions on recompositons.

**Review by G0retZ — CHANGES_REQUESTED** [review]

**Review by GuiFPires — APPROVED** [review]

File: app-games/src/main/java/com/aptoide/android/aptoidegames/appview/AppViewScreen.kt [hunk 8]
```diff
@@ -208,6 +211,8 @@ fun AppViewContent(
   navigate: (String) -> Unit,
   navigateBack: () -> Unit,
 ) {
+  val (bundleTitle, bundleTag) = rememberBonusBundle() ?: (null to null)
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/appview/AppViewScreen.kt:214]: This looks weird.
Better us it without destructor like so:
`val bonusBundle = rememberBonusBundle()`

File: app-games/src/main/java/com/aptoide/android/aptoidegames/appview/AppViewScreen.kt [hunk 9]
```diff
@@ -297,6 +302,13 @@ fun AppViewContent(
         modifier = Modifier
           .padding(top = 160.dp)
           .align(Alignment.TopStart)
+          .clickable {
+            if (bundleTitle != null && bundleTag != null) {
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/appview/AppViewScreen.kt:306]: This looks weird.
Better us it as a single value like so:
`if (bonusBundle != null)`

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/BonusSectionView.kt [hunk 10]
```diff
@@ -47,6 +49,10 @@ fun BonusSectionView(
   bundle: Bundle,
   navigate: (String) -> Unit,
 ) {
+  val bonusViewModel = hiltViewModel<BonusViewModel>()
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/BonusSectionView.kt:52]: Better encapsulate the function into the `rememberBonusBundle` and use it here

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/BonusViewModel.kt [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+  }
+}
+
+@Composable
+fun rememberBonusBundle() = runPreviewable(
+  preview = { null },
+  real = {
+    val vm = hiltViewModel<BonusViewModel>()
+    val bonusInfo by vm.uiState.collectAsState()
+    bonusInfo
```
Comment (G0retZ) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/BonusViewModel.kt:46]: Make a method encapsulation here for wider usage
`bonusInfo to bonusViewModel.addBundleData`

**Review by G0retZ — CHANGES_REQUESTED** [review]

File: feature-bonus/build.gradle.kts
Comment (GuiFPires) [inline feature-bonus/build.gradle.kts:1]: Missing EOF empty line

File: feature-bonus/.gitignore
Comment (GuiFPires) [inline feature-bonus/.gitignore:1]: Missing EOF empty line

**Review by GuiFPires — CHANGES_REQUESTED** [review]

**Review by G0retZ — APPROVED** [review]

**Pull request merged**